srcdeps-maven-local-repository/src/main/java/org/srcdeps/mvn/localrepo/SrcdepsLocalRepositoryManager.java

```diff
@@ -161,26 +89,50 @@ public Path getConfigurationLocation() {
         return Collections.unmodifiableList(result);
     }
 
+    private final BuildDirectoriesManager buildDirectoriesManager;
+
     private final BuildService buildService;
 
-    private volatile ConfigurationHolder configurationHolder;
-    private final Object configurationLock = new Object();
+    private final Configuration configuration;
+    private final Path configurationLocation;
+
     private final LocalRepositoryManager delegate;
-    private final Path scrdepsDir;
-    private final BuildDirectoriesManager buildDirectoriesManager;
 
-    private final Provider<MavenSession> sessionProvider;
+    private final Path scrdepsDir;
 
-    public SrcdepsLocalRepositoryManager(LocalRepositoryManager delegate, Provider<MavenSession> sessionProvider,
-            BuildService buildService, PathLocker<SrcVersion> pathLocker) {
+    public SrcdepsLocalRepositoryManager(LocalRepositoryManager delegate, BuildService buildService, PathLocker<SrcVersion> pathLocker) {
         super();
         this.delegate = delegate;
         this.buildService = buildService;
         this.scrdepsDir = delegate.getRepository().getBasedir().toPath().getParent().resolve("srcdeps");
-        this.sessionProvider = sessionProvider;
         this.buildDirectoriesManager = new BuildDirectoriesManager(scrdepsDir, pathLocker);
-    }
 
+        Path basePath = null;
+        try {
+            basePath = new File(System.getProperty(maven_multiModuleProjectDirectory_PROPERTY, ""))
+                    .getCanonicalFile().toPath();
+        } catch (IOException e1) {
+            basePath = new File(System.getProperty(maven_multiModuleProjectDirectory_PROPERTY, ""))
```

Comment:
> Can't this throw an IOException as well?

No, getAbsoluteFile does not throw any checked exception.

If get*Canonical*File throws and NPE, I guess, the whole method will still work, because cannonical vs. absolute is just two ways of accessing the same location. I guess, `getCanonicalFile()` throws the IOE because there is some real FS access there, while `getAbsoluteFile` does just some string ops like concat and replace that do not throw an IOE.

If you ask why I did it like that, I must admit I copied it without too much thinking from https://github.com/apache/maven/blob/master/maven-embedder/src/main/java/org/apache/maven/cli/MavenCli.java#L365-L374

You are right that the code looks stupid. Let me think what can be done about it.